src/instrument/line_coverage.c

```diff
@@ -308,12 +306,11 @@ void MVM_line_coverage_report(MVMThreadContext *tc, MVMString *filename, MVMuint
     if (tc->instance->coverage_control == 2 || (!tc->instance->coverage_control && cache[cache_slot] == 0)) {
         char *encoded_filename;
         char composed_line[256];
-        size_t length;
 
         cache[cache_slot] = 1;
 
         encoded_filename = MVM_string_utf8_encode_C_string(tc, filename);
-        if ((length = snprintf(composed_line, 255, "HIT  %s  %d\n", encoded_filename, line_number)) > 0) {
+        if (snprintf(composed_line, 255, "HIT  %s  %d\n", encoded_filename, line_number) > 0) {
```

Comment:
Whilst this change looks valid, wouldn't be even better to replace `snprintf`-then-`fputs` with a single `fprintf` ?

(I'm only feeling half awake - am I missing something?)

Reply:
Sure, that would make sense, but I'm not sure why we limit the line to 255 characters. Ah, https://github.com/MoarVM/MoarVM/commit/80e5fe5ad9cbb643d1579515e8e66aa462c691ee changes it from fprintf to snprintf+fputs. @timo I don't understand what that commit message means? And why didn't you use `%.255s` with the fprintf instead? Is it not portable?

Reply:
I got curious and went to look at the glibc source and at least there it [seems to acquire a lock](https://github.com/lattera/glibc/blob/master/stdio-common/vfprintf.c#L1318). The [docs for flockfile](https://man7.org/linux/man-pages/man3/flockfile.3.html) indicate that, with few exceptions, the stdio functions applying locking. So in theory a single `printf` would do it.